### ISSUE TITLE
Incorrect emit of class extend when scoping a lot

### ISSUE THREAD

**Issue by ceymard** [Microsoft/TypeScript#16638]
Note that the title of this bug is not necessarily a good one.

**TypeScript Version:**  2.4.0

**Code**

```ts
class Test { }

function test() {       
                
        if (true) {
                class Test2 extends Test { }

                return [1, 2, 3].map(b => {
                        var a = new Test2()
                        return a
                })

        }
}
```

**Expected behavior:**

I'm going to make it short, I expect Test2 to look like

```js
var Test2_1 = (function (_super) {
            __extends(Test2 /** here **/, _super);
            function Test2() {
                return _super !== null && _super.apply(this, arguments) || this;
            }
            return Test2;
        }(Test));
```

**Actual behavior:**

Instead, Test2 is inlined as Test2_1 inside the function

```js
var Test2_1 = (function (_super) {
            __extends(Test2_1 /** here **/, _super);
            function Test2() {
                return _super !== null && _super.apply(this, arguments) || this;
            }
            return Test2;
        }(Test));
```

This problem does not occur if the `if` is removed or the inline map inside the if.

**Comment by ceymard** [comment]
I will just add that this problem only occurs when `Test2`'s name is I suppose guarded against potential conflicts by renaming it to `Test2_1`.

As long as `Test2` stays the same, there is no problem.

I suppose this is because `__extends(<name>, ...)`  is using the outside name while it should have been using the one inside instead.

**Comment by mhegazy** [comment]
Should be fixed in 2.6
